### ISSUE TITLE
kafka python driver creates warnings and errors in shotover

### ISSUE THREAD

**Issue by rukai** [shotover/shotover-proxy#1579]
When python producer terminates it generates this log in shotover (for whichever node it was connected to at the time i think:

 17 03:51:10 i495b4929 shotover-proxy[147000]: 2024-04-17T03:51:10.530285Z ERROR shotover::server: connection was unexpectedly terminated
Apr 17 03:51:10 i495b4929 shotover-proxy[147000]: Caused by:
Apr 17 03:51:10 i495b4929 shotover-proxy[147000]:     0: Chain failed to send and/or receive messages, the connection will now be closed.
Apr 17 03:51:10 i495b4929 shotover-proxy[147000]:     1: KafkaSinkCluster transform failed
Apr 17 03:51:10 i495b4929 shotover-proxy[147000]:     2: Invalid metadata, controller points at unknown node BrokerId(-1)

While the consumer is connected it generates this warning fairly regularly in the shotover node it's connected to:

Apr 17 03:56:02 i495b4929 shotover-proxy[147000]: 2024-04-17T03:56:02.185301Z  WARN shotover::server: failed to receive message on tcp stream: Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" }
Apr 17 03:56:13 i495b4929 shotover-proxy[147000]: 2024-04-17T03:56:13.488846Z  WARN shotover::server: failed to receive message on tcp stream: Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" }
Apr 17 03:56:19 i495b4929 shotover-proxy[147000]: 2024-04-17T03:56:19.544770Z  WARN shotover::server: failed to receive message on tcp stream: Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" }
Apr 17 03:56:20 i495b4929 shotover-proxy[147000]: 2024-04-17T03:56:20.200930Z  WARN shotover::server: failed to receive message on tcp stream: Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" }
Apr 17 03:56:46 i495b4929 shotover-proxy[147000]: 2024-04-17T03:56:46.339697Z  WARN shotover::server: failed to receive message on tcp stream: Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" }

the messages do get published and are able to be consumed though from what i can see

**Comment by justinweng-instaclustr** [comment]
Tried to reproduce the error log `shotover::server: connection was unexpectedly terminated` and found it only happens when Kafka brokers are killed, and doesn't happen when the Kafka Python producer is terminated.

When the Python producer is abruptly terminated:

<img width="1642" height="671" alt="Image" src="https://github.com/user-attachments/assets/793b8d8b-f52a-4189-bb45-cf456da3a087" />


There's no Error or Warning logs:

<img width="1239" height="952" alt="Image" src="https://github.com/user-attachments/assets/6dd3735a-9ec4-4650-903c-5966fb819428" />

When the Java producer is abruptly terminated:

<img width="1700" height="589" alt="Image" src="https://github.com/user-attachments/assets/3d483662-10f8-4285-b9dd-3a1bf79fac12" />

Also no Error or Warning logs:

<img width="1159" height="933" alt="Image" src="https://github.com/user-attachments/assets/910466f8-b338-4312-a36e-7f4bfc1b7bbe" />


This `shotover::server: connection was unexpectedly terminated` log could be successfully reproduced by killing the Kafka broke (i.e., go to the Kafka node and run `sudo systemctl stop kafka`).

<img width="1083" height="239" alt="Image" src="https://github.com/user-attachments/assets/8dd82167-f992-4972-82c0-93ccb92e3ba0" />

But in this case I think we expect to see this error log.

The warning log of `WARN shotover::server: failed to receive message on tcp stream: Os` couldn't be reproduced at all.

The integration tests implemented by Cursor did reproduce both the Error log `shotover::server: connection was unexpectedly terminated` and the Warning log `WARN shotover::server: failed to receive message on tcp stream: Os`. It also kills Kafka broker service to reproduce the error (which should be an expected error in that case) and it abruptly disconnect the python client by TCP RST to reproduce the Warning log.

In summary, I think we don't change anything for that Error log and we can remove that warning log if we don't want noisy logs in shotover due to client disconnection.
